src/core/config/ip6.h

```diff
@@ -237,6 +237,16 @@
 #define OPENTHREAD_CONFIG_IP6_RESTRICT_FORWARDING_LARGER_SCOPE_MCAST_WITH_LOCAL_SRC 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR
+ *
+ * Define as 1 to handle ICMP echo request destined for locator addresses.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR
+#define OPENTHREAD_CONFIG_IP6_HANDLE_ECHO_REQUEST_DESTINED_FOR_LOCATOR 1
+#endif
```

Comment:
The change in this PR LGTM. Thanks. 👍 

Did a bit of background, looks like the change to `// always handle Echo Request destined for RLOC or ALOC` was added way back in https://github.com/openthread/openthread/pull/3011.

One idea regarding this:
- We have `otIcmp6EchoMode` which allows users to set the behavior regarding handling of echo requests by OT stack.
- We can consider adding the new behavior (handling pings for RLOC/ALOC) as a new option in this `enum`?
- This way, we have one way of configuring how pings are handled.

Maybe the following:

```c++
typedef enum otIcmp6EchoMode
{
    OT_ICMP6_ECHO_HANDLER_DISABLED       = 0, ///< Disabled
    OT_ICMP6_ECHO_HANDLER_UNICAST_ONLY   = 1, ///< Enabled only for unicast requests only
    OT_ICMP6_ECHO_HANDLER_MULTICAST_ONLY = 2, ///< Enabled only for multicast requests only
    OT_ICMP6_ECHO_HANDLER_ALL            = 3, ///< Enabled for unicast and multicast requests
    OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY = 4, ///< Enabled for RLOC/ALOC destinations only
} otIcmp6EchoMode;
```

Thoughts?

Reply:
Thanks for the comment, checked with @librasungirl that #3011 was for wpantund. Do you know if wpantund should continue to be supported @jwhui ? or can we simply remove the special handling for rloc/aloc in #3011?

Reply:
Thanks @abtink for the suggestion, I updated the code to switch from using new flag to using new enum in `otIcmp6EchoMode`. The posix code sets the mode to `OT_ICMP6_ECHO_HANDLER_RLOC_ALOC_ONLY` to keep the existing behavior, wpantund would be supported if it's used by anyone. For systems that want to disable echo handling from OT completely they can set the mode to `OT_ICMP6_ECHO_HANDLER_DISABLED`.